src/coinbase/types.ts

```diff
@@ -1362,3 +1362,20 @@ export interface SmartContractAPIClient {
     options?: RawAxiosRequestConfig,
   ): AxiosPromise<SmartContractModel>;
 }
+
+/**
+ * Options for pagination on list methods.
+ */
+export type PaginationOptions = {
+  limit?: number;
+  page?: string | undefined;
+};
+
+/**
+ * Paginated list response.
+ */
+export interface PaginationResponse<T> {
+  data: T[];
+  hasMore: boolean;
```

Comment:
if `hasMore` is true, `nextPage` is always set right? if so, i think this response type could be improved because it currently allows for states where `hasMore` is true and nextPage is undefined, so users would have to write code like:
```
if (hasMore && !nextPage) { 
  fetch(nextPage)
}
```
OR

```
if (hasMore) {
  fetch(nextPage!)  // suppress the undefined warning
}
```

Reply:
+1 there may be an ergonomics improvement we can make with composite types. As long as that will remain compatible with the current interface, let's take this as a fast follow to investigate further.